lightningd/pay.c

```diff
@@ -913,17 +919,20 @@ static void json_sendpay_on_resolve(const struct sendpay_result* r,
 static void json_sendpay(struct command *cmd,
 			 const char *buffer, const jsmntok_t *params)
 {
-	const jsmntok_t *routetok, *rhashtok;
+	const jsmntok_t *routetok, *rhashtok, *desctok;
 	const jsmntok_t *t, *end;
 	size_t n_hops;
 	struct sha256 rhash;
 	struct route_hop *route;
 	u64 *msatoshi;
+	const struct json_escaped *desc;
+	const char *description;
 
 	if (!param(cmd, buffer, params,
 		   p_req("route", json_tok_tok, &routetok),
 		   p_req("payment_hash", json_tok_tok, &rhashtok),
 		   p_opt("msatoshi", json_tok_u64, &msatoshi),
+		   p_opt_tok("description", &desctok),
```

Comment:
Just a side note: Despite all our efforts with the new parameter parsing system, we still can't directly capture a simple string.  I'm open to any ideas.

Reply:
There is `json_tok_escaped_string` which does an ok job imho

Reply:
Hmm, we could add a ctx arg to all json_tok_() and then pass cmd in, then make json_tok_escaped_string of form `bool (const tal_t *ctx, const char *buffer, const jsmntok_t *tok, char **str)` to match?

Reply:
Yes that would be helpful in many places, including `json_tok_label`. It would go a long way in helping us remove usage of `json_tok_tok`.